From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "David C. Manuelda" <StormByte@gmail.com>
Date: Fri, 4 Apr 2025 21:24:13 +0200
Subject: [PATCH] boot: only link libgcc when compiler is gcc

When compiling in systems which do not have gcc installed
(like a musl+llvm system) the forced linkage "-lgcc" is
stopping it to compile. As when compiler is clang it do not
need to link explicitelly to gcc I've modified meson to only
link to gcc library when compiler is gcc.

Closes: #35162
---
 src/boot/meson.build | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/boot/meson.build b/src/boot/meson.build
index 9370274d96..28eed9b36b 100644
--- a/src/boot/meson.build
+++ b/src/boot/meson.build
@@ -152,7 +152,6 @@ if get_option('mode') == 'developer' and get_option('debug')
 endif
 
 efi_c_ld_args = [
-        '-lgcc',
         '-nostdlib',
         '-static-pie',
         '-Wl,--entry=efi_main',
@@ -176,6 +175,11 @@ efi_c_ld_args = [
         '-z', 'separate-code',
 ]
 
+# Check if the compiler is GCC and then we link to lgcc in that case only
+if cc.get_id() == 'gcc'
+    efi_c_ld_args += ['-lgcc']
+endif
+
 efi_c_ld_args += cc.get_supported_link_arguments(
         # binutils >= 2.38
         '-Wl,-z,nopack-relative-relocs',
